rfc/development-cycle.md

```diff
@@ -0,0 +1,180 @@
+
+# Jupyter Development Cycle
+
+The goal of this document is to outline a development cycle for JupyterLab which makes it very easy to do releases. As [documented in this issue](https://github.com/jupyterlab/jupyterlab/issues/8195), it's currently a quite laborious and specialized process. If we can make releases easier to manage, and push any of the hard work around so it's tackled by the authors of the PRs instead of the release managers, we can release more easily and free up core developer time for other matters.
+
+This is meant to support our existing release cycles, not change them, but just provide some standardization and tooling to aid us in keeping to them.
+
+## Assumptions
+
+I have found it helpful to try to first outline the assumptions of our release and development process.
+
+Please give feedback on these first.
+
+You can consider these in addition to the [SemVer 2.0.0](https://semver.org/) spec, having more to do with the relations between the code evolution between multiple versions. They assume we are already following SemVer.
+
+
+**alpha, beta, rc**: Each release goes through some optional number of alpha, beta, and RC release, before the final. For some final release `{x}.{y}.{z}` it can be modeled with this state machine
+
+1. Start by releasing one of these:
+    1. `{x}.{y}.{z}a0`
+    2. `{x}.{y}.{z}b0`
+    3. `{x}.{y}.{z}rc0`
+    4. `{x}.{y}.{z}`
+2. Then the next release can be either an increment of the last number or a release of one of the initial following your
+   release. For example after `{x}.{y}.{z}b10` you could release `{x}.{y}.{z}b11` or `{x}.{y}.{z}rc0` or `{x}.{y}.{z}` (final).
+
+**Previous Ancestor**: The next increment will be the git ancestor of the previous increment.
+
+* Patch: `{x}.{y}.{z}` is an ancestor of `{x}.{y}.{z+1}`.
+* Minor: `{x}.{y}.0` is an ancestor of `{x}.{y+1}.0`.
+* Major: `{x}.0.0` is an ancestor of `{y+1}.0.0`.
+
+**Single Active**: No commits will be queued up for a release until the previous increment was fully release
+(this also falls out of the above assumption). 
+
+* Patch: No commits towards `{x}.{y}.{z}` till `{x}.{y}.{z-1}` final is released.
+* Minor: No commits towards `{x}.{y}.0` till `{x}.{y-1}.0` final is released.
+* Major: No commits towards `{x}.0.0` till `{x-1}.0.0` final is released.
+
+The last two mean you have to totally finish release before you start on the next one. For example, it would be illegal to have releases in this order: `1.2.0a0`, `1.3.0a0`, `1.2.0`, then `1.3.0`.
```

Comment:
Where does `1.2.1` fall into this pattern?  On a separate branch?

Reply:
I see below where you answer this question, but I think it would be good to be explicit that you mean for a single branch here.